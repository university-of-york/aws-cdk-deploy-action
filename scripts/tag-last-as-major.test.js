const execa = require('execa');
jest.mock('execa');

test('should execute git commands', async () => {
    process.env.GITHUB_TOKEN = '123';
    process.env.GITHUB_REPOSITORY = 'some-name';

    require('./tag-last-as-major'); // eslint-disable-line import/no-unassigned-import
    // setting a delay is required to get the mock output of each function
    await new Promise((resolve) => setTimeout(resolve, 1000));
    expect(execa.mock.calls[0]).toMatchSnapshot();
    expect(execa.mock.calls[1]).toMatchSnapshot();
    expect(execa.mock.calls[2]).toMatchSnapshot();
});
