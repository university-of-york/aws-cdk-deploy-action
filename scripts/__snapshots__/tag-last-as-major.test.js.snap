// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should execute git commands 1`] = `
Array [
  "git",
  Array [
    "push",
    "https://x-access-token:123@github.com/some-name.git",
    "--delete",
    "v1",
  ],
]
`;

exports[`should execute git commands 2`] = `
Array [
  "git",
  Array [
    "tag",
    "-f",
    "v1",
  ],
]
`;

exports[`should execute git commands 3`] = `
Array [
  "git",
  Array [
    "push",
    "https://x-access-token:123@github.com/some-name.git",
    "main",
    "--tags",
  ],
]
`;
